client.go

```diff
@@ -150,17 +145,8 @@ func newClient() (*client, error) {
 
 // Close is used to cleanup the client
 func (c *client) Close() error {
-	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
-	if c.closed {
-		return nil
-	}
-	c.closed = true
-
-	log.Printf("[INFO] mdns: Closing client %v", *c)
+	log.Printf("[INFO] mdns: Closing client: %v", *c)
 	close(c.closedCh)
```

Comment:
Agreed. For now it's only called once from the query() function, which is why this works fine for now. Not sure if there's a way to enforce this in a simple manner.